pkg/controller/enterprisesearch/config.go

```diff
@@ -81,25 +83,36 @@ func ReconcileConfig(client k8s.Client, ents entsv1beta1.EnterpriseSearch) (*cor
 	return reconciledConfigSecret, nil
 }
 
-func newConfig(c k8s.Client, ents entsv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
-	cfg := defaultConfig(ents)
-
+// newConfig builds a single merged config from:
+// - ECK-managed default configuration
+// - association configuration (eg. ES credentials)
+// - TLS settings configuration
+// - user-provided plaintext configuration
+// - user-provided secret configuration
+// In case of duplicate settings, the last one takes precedence.
+func newConfig(c k8s.Client, dynamicWatches watches.DynamicWatches, ents entsv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
```

Comment:
Will follow-up with a PR with unit tests for the config reconciliation once it's complete.
cc @barkbay